xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/email/AbstractMimeMessageIterator.java

```diff
@@ -198,32 +215,59 @@ private void onPrepare(ExtendedMimeMessage message, boolean delete)
     protected abstract List<CompositeEvent> retrieveCompositeEventList(DocumentReference user)
         throws NotificationException;
 
+    private NotificationEmailGroupingStrategy getEmailGroupingStrategy()
```

Comment:
I'm wondering if it would not be better to move all that in a NotificationConfiguration#getEmailGroupingStrategy() since I assume it's not the only place where we'll need to get the NotificationEmailGroupingStrategy component.